### PR TITLE
[+] Transaction table implementation

### PR DESCRIPTION
**Pull request by MiroMargineanu**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "license": "GPL-3.0-or-later",
   "repository": "@elrondnetwork/dapp-template",
   "dependencies": {
-    "@elrondnetwork/dapp-core": "2.0.0",
+    "@elrondnetwork/dapp-core": "2.1.0-rc1",
```
Comment (arhtudormorar) [inline package.json:9]: can upgrade to rc3

File: package.json [hunk 2]
```diff
@@ -75,6 +67,8 @@
     "eslint-plugin-react": "7.30.1",
     "eslint-plugin-react-hooks": "4.6.0",
     "prettier": "2.7.1",
-    "react-error-overlay": "6.0.9"
+    "react-scripts": "4.0.3",
```
Comment (arhtudormorar) [inline package.json:70]: ` "react-scripts": "4.0.3"` should be used in development while `"react-scripts": "5.0.1"` can stay on master

File: src/pages/Dashboard/index.tsx [hunk 3]
```diff
@@ -1,26 +1,67 @@
...
+    loading: false,
+    fetched: false
+  });
+
+  const { account } = useGetAccountInfo();
+  const { network } = useGetNetworkConfig();
+
+  const fetchTransactions = useCallback(() => {
+    const fetchData = async () => {
+      const endpoint = `${network.apiAddress}/accounts/${account.address}/transactions`;
```
Comment (arhtudormorar) [inline src/pages/Dashboard/index.tsx:34]: use ACCOUNTS endpoint from dapp-core enpoints

File: src/pages/Dashboard/index.tsx [hunk 4]
```diff
@@ -1,26 +1,67 @@
...
+      const { data } = await axios.get<ServerTransactionType[]>(endpoint, {
+        params: { size: 150 }
```
Comment (arhtudormorar) [inline src/pages/Dashboard/index.tsx:36]: can get latest 15 only

File: src/pages/Dashboard/index.tsx [hunk 5]
```diff
@@ -1,26 +1,67 @@
...
+        fetched: true
+      });
+    };
+
+    if (!transactions.fetched) {
+      setTransactions((transactions) => ({ ...transactions, loading: true }));
+      fetchData();
+    }
+  }, [
+    account.address,
```
Comment (arhtudormorar) [inline src/pages/Dashboard/index.tsx:51]: can only be account object

File: src/pages/Dashboard/index.tsx [hunk 6]
```diff
@@ -1,26 +1,67 @@
...
-                </div>
-              </div>
-              <Transactions />
-            </div>
-          </div>
-        </div>
-      </div>
+    <div className={`container ${styles.transactions}`}>
+      <TransactionsTable
+        transactions={transactions.data as InterpretedTransactionType[]}
```
Comment (arhtudormorar) [inline src/pages/Dashboard/index.tsx:62]: no need for casting

Comment (arhtudormorar) [reply src/pages/Dashboard/index.tsx:62]: show a loader from dapp-core while loading

File: src/pages/Dashboard/index.tsx [hunk 7]
```diff
@@ -1,26 +1,67 @@
...
+        address={account.address}
```
Comment (arhtudormorar) [inline src/pages/Dashboard/index.tsx:63]: address should no longer be needed

**Review by CiprianDraghici — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (MiroMargineanu) [reply package.json]: Upgraded. 👍

File: package.json (hunk 2 above)
Comment (MiroMargineanu) [reply package.json]: Upgraded. 👍

File: src/pages/Dashboard/index.tsx (hunk 3 above)
Comment (MiroMargineanu) [reply src/pages/Dashboard/index.tsx]: Changed. 👍

File: src/pages/Dashboard/index.tsx (hunk 4 above)
Comment (MiroMargineanu) [reply src/pages/Dashboard/index.tsx]: Changed. 👍

File: src/pages/Dashboard/index.tsx (hunk 5 above)
Comment (MiroMargineanu) [reply src/pages/Dashboard/index.tsx]: Changed. 👍

File: src/pages/Dashboard/index.tsx (hunk 6 above)
Comment (MiroMargineanu) [reply src/pages/Dashboard/index.tsx]: Added. 👍

File: src/pages/Dashboard/index.tsx (hunk 7 above)
Comment (MiroMargineanu) [reply src/pages/Dashboard/index.tsx]: Removed. 👍

**Review by arhtudormorar — APPROVED** [review]

**Pull request merged**